ooniprobe/Utility/TestUtility.mm

```diff
@@ -144,7 +144,23 @@ + (void)downloadUrlsCallback:(NSData *)data
     successcb(urls);
 }
 
-+ (void)removeFile:(NSString*)fileName {
++ (void)deleteUploadedJsons:(void (^)(Measurement*))successcb onError:(void (^)(NSError*))errorcb {
+    for (Measurement *measurement in [Measurement measurementsWithJson]){
+        [measurement getExplorerUrl:^(NSString *measurement_url){
+            [TestUtility removeFile:[measurement getReportFile]];
+            successcb(measurement);
+        } onError:^(NSError *error){
+            errorcb(error);
+        }];
+    }
+}
```

Comment:
So, here's an alternative idea. Let me know what are your thoughts:

```ObjectiveC

+ (void)deleteUploadedJsonsWithMeasurementRemover:(void (^)(Measurement *))remover {
    for (Measurement *measurement in [Measurement measurementsWithJson]) {
        [measurement getExplorerUrl:^(NSString *measurement_url){
            remover(measurement);
        } onError:nil];
    }
}

+ (void)deleteUploadedJsons{
    [self deleteUploadedJsonsWithMeasurementRemover:^(NSString *path) {
        [TestUtility removeFile:[measurement getReportFile]];
    }];
}
```

Then:

1. in the callback in the tests you assert that you only see the measurements you expected to see

2. you use the expectation pattern with a `60` seconds timeout and declare the test failed if we are not able to walk through the measurements in such timeout

This way you don't need to be concerned about counting and you don't need to be concerned about knowing when you have walked through all the list. You only know it's reasonable for it to finish in `60` seconds.

(It's probably a bit of a hack, but my current level of ObjectiveC-fu is such that I cannot think of anything better that does not also make code more complex. If we had coroutines, this would be simpler.)

Reply:
Can you write the test even as pseudo code? 
I don't see what exactly you are suggesting here.
Are you suggesting that we don't actually remove the json, only to be sure that functions return the callback on the two measurements we are expecting to delete?

Reply:
Yes, I am suggesting that we just check that it the callback is called with the right measurements. This is more a unit test than an integration test, in that we do not verify that files are actually deleted. An integration test may perhaps be written later.

We can also write another test for the function deleting files, maybe not in this specific file tho.

Reply:
Did it but I still have some question, check the comment

Reply:
Ok